app/models/manageiq/providers/cloud_manager/orchestration_stack.rb

```diff
@@ -16,4 +19,13 @@ def self.create_stack(orchestration_manager, stack_name, template, options = {})
                  :cloud_tenant           => tenant,
                  :orchestration_template => template)
   end
+
+  def self.orchestration_stack_class_factory(orchestration_manager, template)
+    case template
+    when OrchestrationTemplateVnfd
+      orchestration_manager.class::Vnf
+    else
+      orchestration_manager.class::OrchestrationStack
+    end
```

Comment:
I wonder if the template itself could indicate the type of supported `stack` class.  So, the `OrchestrationTemplateVnfd` template class would return `Vnf` and all others would return `OrchestrationStack`.

Might need to pass in the `manager_class` to fully qualify the returned class name, but something along those lines might remove the special handling logic from here.

/cc @bzwei 

Thoughts?